llvm/lib/Passes/PassBuilderPipelines.cpp

```diff
@@ -1629,6 +1629,10 @@ PassBuilder::buildFatLTODefaultPipeline(OptimizationLevel Level, bool ThinLTO,
     MPM.addPass(buildLTOPreLinkDefaultPipeline(Level));
   MPM.addPass(EmbedBitcodePass(ThinLTO, EmitSummary));
 
+  // If we're doing FatLTO w/ CFI enabled, we don't want the type tests in the
+  // object file.
+  MPM.addPass(LowerTypeTestsPass(nullptr, nullptr, true, true));
```

Comment:
How does this build? The ctor only takes three arguments.

Reply:
Oh I see, you have a dependent change that adds the argument.

https://github.com/llvm/llvm-project/commit/68a064b39b86d69c02de09cda2f06f4b87c2e69a

I'm not really a big fan of adding another bool argument like this because it makes the callers less clear, can you do something else like make the existing argument into an enum?

Reply:
Agreed. I'm not super happy w/ the boolean either. The only reason its required is to avoid asserting for the new use case, but that also implies that assertion doesn't always hold, so I'm not sure if its really a good idea to keep it.

Reply:
And apologies for not adding you on both reviews. I had thought I added you and Nikita to https://github.com/llvm/llvm-project/pull/112787, but it looks like I hadn't.

Reply:
The assertion is to check that the code is only removing type test assumes (possibly involving a phi). So I wouldn't call the functionality that you're adding "always drop type tests". What you're adding is more like "drop type tests" and what's there already is "drop type test assumes". So maybe you can name the enumerators based on that.